Documentation/RuleDocumentation.md

```diff
@@ -59,6 +61,17 @@ Lint: If a public declaration is missing a documentation comment, a lint error i
 
 `AllPublicDeclarationsHaveDocumentation` is a linter-only rule.
 
+### AlwaysUseLiteralForEmptyCollectionInit
+
+Never use `[<Type>]()` syntax. In call sites that should be replaced with `[]`,
+for initializations use explicit type combined with empty array literal `let _: [<Type>] = []`
+Static properties of a type that return that type should not include a reference to their type.
+
+Lint:  Non-literal empty array initialization will yield a lint error.
+Format: All invalid use sites would be related with empty literal (with or without explicit type annotation).
```

Comment:
"would be _replaced_ with empty literal..."?

Reply:
Ah, that's a typo from earlier, there was a race between the generate-the-docs PR and the one adding this rule so it didn't get generated the first time. I'll update that in a separate follow-up because I need to do a pass over all the rule docs anyway to make sure they're formatted nicely in the new Markdown file.